src-tauri/src/tray/tray.rs

```diff
@@ -0,0 +1,148 @@
+};
+
+pub fn update_tray_menu(state: State<'_, AppState>) -> Result<()> {
+    let app_handle = state.app_handle.clone();
+    let nodes = get_nodes(state)?;
+    let menu = match nodes.len() {
+        0 => build_empty_node_menu(),
+        1 => build_single_node_menu(&nodes[0].name),
+        _ => build_multi_node_menu(&nodes),
+    };
+
+    app_handle.tray_handle().set_menu(menu)?;
+    Ok(())
+}
+
+fn build_single_node_menu(node: &str) -> SystemTrayMenu {
+    let is_running = is_node_process_running(node);
+    let status_icon = if is_running { "🟢" } else { "🔴" };
+
+    let mut menu = SystemTrayMenu::new();
+    menu = menu
+        .add_item(CustomMenuItem::new(
+            format!("show_{}", node),
+            format!("{} {}", status_icon, node),
+        ))
+        .add_native_item(SystemTrayMenuItem::Separator);
+
+    menu = add_node_items(menu, node, is_running);
+
+    menu.add_native_item(SystemTrayMenuItem::Separator)
+        .add_item(CustomMenuItem::new("quit".to_string(), "Quit"))
+}
+
+fn build_empty_node_menu() -> SystemTrayMenu {
+    let no_nodes_item = CustomMenuItem::new("show_window".to_string(), "No nodes available");
+    SystemTrayMenu::new()
+        .add_item(no_nodes_item)
+        .add_native_item(SystemTrayMenuItem::Separator)
+        .add_item(CustomMenuItem::new("quit".to_string(), "Quit"))
+}
+
+fn build_multi_node_menu(nodes: &Vec<NodeInfo>) -> SystemTrayMenu {
+    let mut menu = SystemTrayMenu::new();
+
+    for node in nodes {
+        let is_running = is_node_process_running(&node.name);
+        let status_icon = if is_running { "🟢" } else { "🔴" };
+        let node_menu = add_node_items(SystemTrayMenu::new(), &node.name, is_running);
+        menu = menu.add_submenu(SystemTraySubmenu::new(
+            format!("{} {}", status_icon, node.name),
+            node_menu,
+        ));
+    }
+
+    menu.add_native_item(SystemTrayMenuItem::Separator)
+        .add_item(CustomMenuItem::new("quit".to_string(), "Quit"))
+}
+
+fn add_node_items(menu: SystemTrayMenu, node: &str, is_running: bool) -> SystemTrayMenu {
+    menu.add_item(CustomMenuItem::new(
+        format!("show_{}", node),
+        format!("Show {} window", node),
+    ))
+    .add_item(if is_running {
+        CustomMenuItem::new(format!("start_{}", node), "Start").disabled()
+    } else {
+        CustomMenuItem::new(format!("start_{}", node), "Start")
+    })
+    .add_item(if is_running {
+        CustomMenuItem::new(format!("stop_{}", node), "Stop")
+    } else {
+        CustomMenuItem::new(format!("stop_{}", node), "Stop").disabled()
+    })
+    .add_item(CustomMenuItem::new(format!("config_{}", node), "Configure"))
+    .add_item(CustomMenuItem::new(format!("logs_{}", node), "Logs"))
+    .add_item(CustomMenuItem::new(format!("delete_{}", node), "Delete"))
+}
+
+pub fn handle_tray_click(app_handle: &AppHandle, menu_id: &str) -> Result<()> {
+    let parts: Vec<&str> = menu_id.split('_').collect();
+    match parts.as_slice() {
```

Comment:
no need to allocate

```suggestion
    let parts = menu_id.split('_');
    match [parts.next(), parts.next(), parts.next()] {
```

the last `parts.next()` is a sanity check